test/functional/feature_rbf.py

```diff
@@ -89,29 +96,23 @@ def run_test(self):
     def make_utxo(self, node, amount, confirmed=True, scriptPubKey=DUMMY_P2WPKH_SCRIPT):
         """Create a txout with a given amount and scriptPubKey
 
-        Mines coins as needed.
+        Assumes that MiniWallet has enough funds to cover the amount and the fixed fee
+        (from it's internal utxos, the one with the largest value is taken).
 
         confirmed - txouts created will be confirmed in the blockchain;
                     unconfirmed otherwise.
         """
-        fee = 1 * COIN
-        while node.getbalance() < satoshi_round((amount + fee) / COIN):
-            self.generate(node, COINBASE_MATURITY)
-
-        new_addr = node.getnewaddress()
-        txid = node.sendtoaddress(new_addr, satoshi_round((amount + fee) / COIN))
-        tx1 = node.getrawtransaction(txid, 1)
-        txid = int(txid, 16)
-        i, _ = next(filter(lambda vout: new_addr == vout[1]['scriptPubKey']['address'], enumerate(tx1['vout'])))
-
-        tx2 = CTransaction()
-        tx2.vin = [CTxIn(COutPoint(txid, i))]
-        tx2.vout = [CTxOut(amount, scriptPubKey)]
-        tx2.rehash()
-
-        signed_tx = node.signrawtransactionwithwallet(tx2.serialize().hex())
-
-        txid = node.sendrawtransaction(signed_tx['hex'], 0)
+        # MiniWallet only supports sweeping utxos to its own internal scriptPubKey, so in
+        # order to create an output with arbitrary amount/scriptPubKey, we have to add it
+        # manually after calling the create_self_transfer method. The MiniWallet output's
+        # nValue has to be adapted accordingly (amount and fee deduction). To keep things
+        # simple, we use a fixed fee of 1000 Satoshis here.
+        fee = 1000
+        tx = self.wallet.create_self_transfer(from_node=node, fee_rate=0, mempool_valid=False)['tx']
+        assert_greater_than(tx.vout[0].nValue, amount + fee)
+        tx.vout[0].nValue -= (amount + fee)           # change output -> MiniWallet
```

Comment:
What happens if this output is removed? Previously the wallet would also "donate" the change to the fee.

Reply:
IIUC what you mean: without this output, no new utxo to the internal MiniWallet scriptPubKey is created, i.e. we would probably run out of funds (or, could only create as many transactions as there are coinbase outputs from the pre-mined blocks from the test framework).